src/app/components/cart/components/cart-modal/cart-modal.component.ts

```diff
@@ -0,0 +1,233 @@
+// cart-modal.component.ts
+import {
+  Component,
+  OnInit,
+  OnDestroy,
+  ChangeDetectorRef,
+  ChangeDetectionStrategy,
+  EventEmitter,
+  Output,
+} from '@angular/core';
+import { Nft, Collection, MIN_AMOUNT_TO_TRANSFER } from '@build-5/interfaces';
+import { Subscription, forkJoin, map, take, catchError, of } from 'rxjs';
+import { CartService, CartItem } from './../../services/cart.service';
+import { AuthService } from '@components/auth/services/auth.service';
+import { Router } from '@angular/router';
+import { ROUTER_UTILS } from '@core/utils/router.utils';
+import { NzModalService } from 'ng-zorro-antd/modal';
+import { CheckoutOverlayComponent } from '../checkout/checkout-overlay.component';
+import { NftApi } from '@api/nft.api';
+import { NzNotificationService } from 'ng-zorro-antd/notification';
+
+@Component({
+  selector: 'wen-app-cart-modal',
+  templateUrl: './cart-modal.component.html',
+  styleUrls: ['./cart-modal.component.less'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CartModalComponent implements OnInit, OnDestroy {
+  private subscriptions = new Subscription();
+  public collectionPath: string = ROUTER_UTILS.config.collection.root;
+  public nftPath: string = ROUTER_UTILS.config.nft.root;
+  public cartItemsQuantities: number[] = [];
+  cartItemPrices: { [key: string]: { originalPrice: number; discountedPrice: number } } = {};
+  isCartCheckoutOpen = false;
+
+  constructor(
+    public cartService: CartService,
+    private cd: ChangeDetectorRef,
+    public auth: AuthService,
+    private modalService: NzModalService,
+    private nftApi: NftApi,
+    private notification: NzNotificationService,
+    private router: Router,
+  ) {}
+
+  ngOnInit() {
```

Comment:
note, API supports max 100 qty

Reply:
Ah good to know, will put a limit of 100 NFT IDs in cart, likely good to put a limit on this anyway.  Thanks.